### ISSUE TITLE
Compiler warnings: using 'class' keyword for protocol inheritance is deprecated; use 'AnyObject' instead

### ISSUE THREAD

**Issue by bobergj** [launchdarkly/ios-client#247]
**Is this a support request?**
No

**Describe the bug**

Compiler warnings: "warning: using 'class' keyword for protocol inheritance is deprecated; use 'AnyObject' instead" when linking the LaunchDarkly Swift package.

**To reproduce**
1. Add LaunchDarkly swift package to Xcode project
2. `import LaunchDarkly` somewhere in app target
3. Build the app the app target

**Expected behavior**
No compiler warnings from LaunchDarkly that pollute our build log

**Logs**
```
<censored>/SourcePackages/checkouts/ios-client-sdk/LaunchDarkly/LaunchDarkly/Networking/DarklyService.swift:15:35: warning: using 'class' keyword for protocol inheritance is deprecated; use 'AnyObject' instead

<censored>/checkouts/ios-client-sdk/LaunchDarkly/LaunchDarkly/Networking/DarklyService.swift:22:33: warning: using 'class' keyword for protocol inheritance is deprecated; use 'AnyObject' instead
```

**SDK version**
5.3.2, but the issues occur in the latest code as well:
https://github.com/launchdarkly/ios-client-sdk/blob/v5/LaunchDarkly/LaunchDarkly/Networking/DarklyService.swift#L15
https://github.com/launchdarkly/ios-client-sdk/blob/v5/LaunchDarkly/LaunchDarkly/Networking/DarklyService.swift#L22

**Language version, developer tools**
Xcode Version 12.5 (12E262)

**OS/platform**
macOS 11.2.3 (20D91)

**Additional context**
N/A

**Comment by bwoskow-ld** [comment]
Hi @bobergj,

Thanks for letting us know about this. We haven't yet seen these warnings ourselves and are thinking that this deprecation check may have been introduced in Xcode 12.5.

I have noted this issue on our backlog and we will address it in an upcoming SDK release.

Cheers,
@bwoskow-ld 

_Filed internally as 110317._

**Comment by bobergj** [comment]
> are thinking that this deprecation check may have been introduced in Xcode 12.5.

Yes, this deprecation warning is introduced in Swift 5.4 which shipped in Xcode 12.5

**Comment by gwhelanLD** [comment]
Hi @bobergj,

We've included a fix for this issue in the [5.4.2 release](https://github.com/launchdarkly/ios-client-sdk/releases/tag/5.4.2). Thanks for reporting it!

Thanks,
@gwhelanLD
